packages/side-nav/src/vaadin-side-nav.js

```diff
@@ -97,36 +114,55 @@ class SideNav extends ElementMixin(ThemableMixin(PolylitMixin(LitElement))) {
 
   /** @protected */
   render() {
-    const label = this.querySelector('[slot="label"]');
-    if (label && this.collapsible) {
-      return html`
-        <details ?open="${!this.collapsed}" @toggle="${this.__toggleCollapsed}">${this.__renderBody(label)}</details>
-      `;
+    return html`
+      <button
+        part="label"
+        @click="${this._onLabelClick}"
+        aria-expanded="${ifDefined(this.collapsible ? !this.collapsed : null)}"
+        aria-controls="children"
+      >
+        <slot name="label" @slotchange="${this._onLabelSlotChange}"></slot>
+        <span part="toggle" aria-hidden="true"></span>
+      </button>
+      <ul id="children" part="children" ?hidden="${this.collapsed}" aria-hidden="${this.collapsed ? 'true' : 'false'}">
```

Comment:
should this id `children` be unique? like the `labelId`above to ensure this doesn't interfere in any kind with different screen-reader or parser of the web app?

Reply:
Please note that this element is placed in shadow DOM, and thus the ID is encapsulated within shadow scope. See e.g. [this article](https://coryrylan.com/blog/accessibility-with-id-referencing-and-shadow-dom) that explains it. Unlike `<ul>`, the label element is placed in light DOM and assigned through slot, so its ID must be unique.

Reply:
Thanks for the article! That makes your snapshot tests even more viable to ensure the encapsulation isn't broken by accident. :)